File: Q03/main.cpp
#include<iostream>
#include<cstring>
using namespace std;

int main(){
    char str[100];
    cin>>str;

    int start=0,end=strlen(str)-1;

    while(start<end){
        if(isalnum(str[start])==0){
            start++;
            continue;
        }
        if(isalnum(str[end])==0){
            end--;
            continue;
        }
        if(tolower(str[start])!=tolower(str[end])){
            printf("Not Plaindrome");
            return 0;
        }
        start++;
        end--;
    }
    printf("Palindrome");
    return 0;
}

File: Q03/main.py
str=input()
start=0
end=len(str)-1

while start<end:
    if str[start].isalnum()==False:
        start+=1
        continue
    if str[end].isalnum()==False:
        end-=1
        continue
    if(str[start].lower()!=str[end].lower()):
        print("Not Palindrome")
        break
    start+=1
    end-=1

else:
    print("Palindrome")